rllib/algorithms/appo/appo_learner.py

```diff
@@ -43,14 +44,18 @@ class AppoLearner(ImpalaLearner):
     and `_update_module_kl_coeff()`
     """
 
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
-
-        # We need to make sure kl_coeff are available as framework tensors that are
-        # registered as part of the graph so that upon update the graph can be updated
-        # (e.g. in TF with eager tracing).
-        self.curr_kl_coeffs_per_module = defaultdict(
-            lambda: self._get_tensor_variable(self.hps.kl_coeff)
+    @override(ImpalaLearner)
+    def build(self):
+        super().build()
+
+        # The current kl coefficients per module as tensor variables
+        # (framework specific).
+        self.curr_kl_coeffs_per_module: LambdaDefaultDict[
```

Comment:
Using this new defaultdict subclass avoids us having to tediously override the `add_module` method.